app/controllers/api/maintaining/concept_exercises_controller.rb

```diff
@@ -0,0 +1,87 @@
+module API
+  module Maintaining
+    class ConceptExercisesController < BaseController
+      def create
+        example_filename_regexp = "[^a-zA-Z0-9._-]"
+        exercise_slug_regexp = "[^a-zA-Z0-9-]"
+
+        unless params[:exercise_track].present?
+          return render_error(
+            :missing_exercise_track,
+            "Exercise track is missing"
+          )
+        end
+
+        # TODO: verify/normalize track name
+
+        unless params[:exercise_slug].present?
+          return render_error(
+            :missing_exercise_slug,
+            "Exercise slug is missing"
+          )
+        end
+
+        if params[:exercise_slug] =~ Regexp.new(exercise_slug_regexp)
+          return render_error(
+            :invalid_exercise_slug,
+            "Exercise slug must match #{exercise_slug_regexp}"
+          )
+        end
+
+        unless params[:example_filename].present?
+          return render_error(
+            :missing_example_filename,
+            "Example filename is missing"
+          )
+        end
+
+        if params[:example_filename] =~ Regexp.new(example_filename_regexp)
+          return render_error(
+            :invalid_example_filename,
+            "Example filename must match #{example_filename_regexp}"
+          )
+        end
+
+        uuid = SecureRandom.uuid
+        tmp_path = Pathname.new("/tmp/#{uuid}")
+        branch_name = "#{params[:exercise_track]}/#{params[:exercise_slug]}-#{uuid}"
+
+        p tmp_path
+
+        `git clone https://#{Rails.application.secrets.exercism_bot_credentials}@github.com/exercism-bot/v3.git #{tmp_path}`
+
+        Dir.chdir(tmp_path) {
```

Comment:
Use `do`/`end` for multiline blocks that don't have some method tagged onto the end:

These are the "valid" approaches in my opinion (ie the opinion of our Ruby codebases :))

```ruby
[].each do
# ...
end

[].each { # ... }

[].each { 
}.something_else
```

Reply:
Right! Makes sense. I've updated